tox.ini

```diff
@@ -39,7 +39,7 @@ commands =
 #
 # - _integration_tests/goldens.json_: A set of goldens which should be
 #   generated by hand, or using the utility in bin/doc_extract.  Alternatively,
-#   they could be generated from darglint and touched up by hand to be
+#   they could be generated from darglint2 and touched up by hand to be
 #   made more correct.
 # - _integration_tests/max_golden.json_: Should contain a very large golden,
 #   crafted to be difficult to parse.
```

Comment:
@terrencepreilly, you also wrote in `tox.ini`:
>These tests expect certain test fixtures to be present.  They're not included here because they could substantially increase the size of the repository, and I don't want to worry about licenses, etc.

Are these fixtures something you won't be able to share with me? If so, what would you suggest to use instead, and what's the correct format?

What about `integration_tests/repos/`, what kinds of repositories to put there and how to use them for the integration tests?

Reply:
Oh, there's a tool to help generate them, under `bin/doc_extract`.  The format is encoded [here](https://github.com/terrencepreilly/darglint/blob/abc26b768cd7135d848223ba53f68323593c33d5/bin/doc_extract/static/src/Main.elm#L278).  Sure, I can email it to you, if you'd like.  I didn't include it because I didn't want to handle all the license issues that might come from including them in the repo.  They're mostly from large open source projects like Django and such.

Reply:
If I understand correctly:
- `doc_extract -s repos.txt -o docstrings.json` accepts local Git repo paths in `repos.txt` and writes docstrings and metadata into `docstrings.json`.
- The Elm webapp accepts `docstrings.json` and lets me interactively choose the type of each docstring, discard docstrings, and save the result back into `output.json`.
- `output.json` can then be used for integration tests.

Is this correct @terrencepreilly?

I tried this out, and clicking on radio buttons to toggle the docstring type in the webapp seems to always cause `Google` to be selected, with no JavaScript errors or anything suspicious in the Elm debug log. Is this a known issue or a problem at my end?

Anyway, I would appreciate it if you could e-mail me (my gmail is akaihola) any integration test data you might have.

Reply:
Yeah, that sounds correct.  I'm not sure about the radio button thing, I tried it and it fails for me as well.  Likely, something changed in the intervening years since this was written.  There's sometimes some strange corners around Elm with custom event listeners (and looks like the radio buttons have to use custom radio buttons.)

Reply:
Your last update to the app in 6824be4dbb15c08e8a1975b99520e3282c29e541 was in February 2020. The NumPy docstring type seems to have been added then, along with some changes to the docstring type logic. Also Elm was updated from 0.19.0 to 0.19.1 (which had [come out in Oct 2019](https://elm-lang.org/news)).

Reply:
Thanks for the goldens you e-mailed me @terrencepreilly.

The radio button issue is now tracked in #7.

Reply:
@real-yfprojects I just remembered I have the `goldens.json` Terrence e-mailed to me, see above for description. It's also mentioned in the change log for `1.1.0-alpha`:

>- A (crappy) integration test framework. Test fixtures are ignored in git, since the integration tests are only relevant for local development (and even then, mostly just release). The integration tests are as follows:
>      - `goldens.py`: Tests against goldens for individual docstrings. This attempts to ensure that parsed docstrings always contain the expected sections after development. Goldens are generated using the `doc_extract` utility in the `bin/` folder.

Would that file be useful for you? I don't remember why it wasn't appropriate to make it public in the repository.

Reply:
Yes, thanks. I imagine a lot of work went into preparing the test data. My gmail is _yfprojects_.